### PR TITLE
Remove some dead code / branches

### PR DESCRIPTION
**Pull request by stephentoub**
Some signal amidst the noise in an lgtm report.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Text/UTF32Encoding.cs [hunk 1]
```diff
@@ -1101,10 +1101,7 @@ public override int GetMaxCharCount(int byteCount)
                 charCount /= 2;
             }
 
-            if (charCount > 0x7fffffff)
-                throw new ArgumentOutOfRangeException(nameof(byteCount), SR.ArgumentOutOfRange_GetCharCountOverflow);
-
-            return (int)charCount;
+            return charCount;
```
Comment (GrabYourPitchforks) [inline src/libraries/System.Private.CoreLib/src/System/Text/UTF32Encoding.cs:1104]: This is actually a bug in the original code. On line 1091 above, _charCount_ should be typed as _long_, not _int_.

File: src/libraries/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs [hunk 2]
```diff
@@ -31,13 +31,12 @@ private FileVersionInfo(string fileName)
...
             // we can implement a full-fledged Win32 resource parser; that would also enable support
             // for native Win32 PE files on Unix, but that should also be an extremely rare case.
-            if (!TryLoadManagedAssemblyMetadata())
-            {
-                // We could try to parse Executable and Linkable Format (ELF) files, but at present
-                // for executables they don't store version information, which is typically just
-                // available in the filename itself.  For now, we won't do anything special, but
-                // we can add more cases here as we find need and opportunity.
-            }
+            TryLoadManagedAssemblyMetadata();
```
Comment (tannergooding) [inline src/libraries/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs:34]: nit: Make it explicit that we are ignoring the return value
```suggestion
            _ = TryLoadManagedAssemblyMetadata();
```

File: src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs [hunk 3]
```diff
@@ -209,30 +209,15 @@ protected override void Dispose(bool disposing)
...
-                    {
-                        _zipStreamManager.Dispose();
-                    }
-
-                    if (_zipArchive != null)
-                    {
-                        _zipArchive.Dispose();
-                    }
+                    _contentTypeHelper?.SaveToFile();
+                    _zipArchive?.Dispose();
```
[the remark this answers is not in the document]
Comment (tannergooding) [reply src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs:213]: Ah, I see you removed `: IDisposable` below

**Review by GrabYourPitchforks — APPROVED** [review]
Suggest reverting the UTF32Encoding part of the change since we're just going to end up adding it back anyway once the bug is fixed, but otherwise LGTM!

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/XdrBuilder.cs [hunk 4]
```diff
@@ -1655,7 +1655,7 @@ private static void ParseDtMaxLength(ref uint cVal, object obj, XdrBuilder build
                 builder.SendValidationEvent(SR.Sch_DupDtMaxLength);
             }
 
-            if (!ParseInteger((string)obj, ref cVal) || cVal < 0)
+            if (!ParseInteger((string)obj, ref cVal))
```
Comment (tannergooding) [inline src/libraries/System.Private.Xml/src/System/Xml/Schema/XdrBuilder.cs:1658]: Just to confirm, the intent here wasn't to check that `cValue <= int.MaxValue`, right?

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/XsltOld/NumberAction.cs [hunk 5]
```diff
@@ -509,11 +509,6 @@ private static string Format(ArrayList numberlist, List<FormatInfo?>? tokens, st
             }
             if (groupingSep != null)
             {
-                if (groupingSep.Length > 1)
-                {
-                    // It is a breaking change to throw an exception, SQLBUDT 324367
-                    //throw XsltException.Create(SR.Xslt_CharAttribute, "grouping-separator");
-                }
```
Comment (tannergooding) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/XsltOld/NumberAction.cs:516]: So we wanted to throw here but can't?

What's the behavior then if the `length > 1`?

**Review by tannergooding — APPROVED** [review]
LGTM. Had a couple questions and a small suggestion

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/Preprocessor.cs [hunk 6]
```diff
@@ -1727,10 +1727,8 @@ private void PreprocessComplexType(XmlSchemaComplexType complexType, bool local)
                 SetParent(complexType.ContentModel, complexType); //SimpleContent / complexCotent
                 PreprocessAnnotation(complexType.ContentModel);
 
-                if (complexType.Particle != null || complexType.Attributes != null)
-                {
-                    // this is illegal
-                }
+                // "complexType.Particle != null || complexType.Attributes != null" is illegal
```
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/libraries/System.Private.Xml/src/System/Xml/Schema/Preprocessor.cs:1730]: I don't actually know if the if block that said "this is illegal" is never hit.

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/XdrBuilder.cs (hunk 4 above)
Comment (stephentoub) [reply src/libraries/System.Private.Xml/src/System/Xml/Schema/XdrBuilder.cs:1658]: I'm not sure what the intent was, but it's been this way for a very long time, so I'd be hesitant to add a new condition without other motivation.

**Comment by stephentoub** [comment]
Thanks for reviewing, Tanner and Levi.

**Review by bartonjs — APPROVED** [review]

**Pull request merged**